gnovm/pkg/gnolang/nodes.go

```diff
@@ -1079,14 +1081,16 @@ type FileSet struct {
 	Files []*FileNode
 }
 
-// TODO replace with some more efficient method
-// that doesn't involve parsing the whole file.
-//
-// name could be anything,
-// it's only used to generate better error traces.
-func PackageNameFromFileBody(name string, body string) Name {
-	n := MustParseFile(name, body)
-	return n.PkgName
+// PackageNameFromFileBody extracts the package name from the given Gno code body.
+// The 'name' parameter is used for better error traces, and 'body' contains the Gno code.
+func PackageNameFromFileBody(name, body string) Name {
```

Comment:
Just out of curiosity, why not extract out the package name using a regex, instead of initializing the standard lib to parse the entire file?
Not sure which is quicker without doing benchmarks

Reply:
> why not extract out the package name using a regex

I had the same thought.

But after reading about the mode `PackageClauseOnly`, it seems very optimized to me. So decided to go with it for now. 

> Not sure which is quicker without doing benchmarks

yeah, it's hard to make any comment without doing benchmarks.

Reply:
> But after reading about the mode PackageClauseOnly, it seems very optimized to me

Can you elaborate on this in the context of using a precompiled regex?

> yeah, it's hard to make any comment without doing benchmarks.

Why not write a benchmark to confirm this, since we're having the same thought about a performance boost?

Reply:
I plan to benchmark this out of curiosity (not my priority right now).  I have no plans to use regex (at least in this PR)
Because of a few reasons:
- Writing a regex pattern that can accurately handle all possible (edge)cases can quickly become overly complicated and error-prone. If the Go/Gno source file structure is modified or formatted differently, the regex pattern might no longer work as expected, leading to inaccurate results or runtime errors
- As the Go evolves, maintaining and updating these regex patterns can become a challenging task.
- My last reason, Go provides a built-in parser package, and the mode which fits our use case (`PackageClauseOnly`), which is designed explicitly for parsing Go code and handling all the complexities of the language's syntax. (and we using the same package for parsing Gno file)